File: WebAPI/Controllers/RulesController.cs
using AutoMapper;
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Threading.Tasks;
using System.Web.Http;
using System.Web.Http.Cors;
using WebAPI.DTOs;
using WebAPI.Models;

namespace WebAPI.Controllers
{
    [EnableCors(origins: "*", headers: "*", methods: "*")]
    public class RulesController : ApiController
    {
        private ApplicationDbContext _context = new ApplicationDbContext();

        public IEnumerable<RuleDTO> Get()
        {
            var rules = _context.Rules.ToList();

            return Mapper.Map<List<Rule>, List<RuleDTO>>(rules);
        }

        public async Task<IHttpActionResult> Post(NewRuleDTO ruleDto)
        {
            if (ruleDto == null)
            {
                return BadRequest("Argument Null");
            }

            var rule = new Rule()
            {
                Move = ruleDto.Move,
                Kills = ruleDto.Kills
            };

            _context.Rules.Add(rule);

            await _context.SaveChangesAsync();

            return Ok();
        }

        public async Task<IHttpActionResult> Delete(int id)
        {
            var rule = await _context.Rules.FirstOrDefaultAsync(r => r.Id == id);

            if (rule == null)
            {
                return NotFound();
            }

            _context.Rules.Remove(rule);

            await _context.SaveChangesAsync();

            return Ok();
        }

    }
}

File: Web/app/controllers/rulesController.ts
module TestApp.Controllers {
    export class RulesController {

        public rules: Array<Models.Rule>;

        public move: string;
        public kills: string;

        static $inject = [Constants.Services.RulesService];
        constructor(
            public rulesService: Services.RulesService
        ) {
            this.load();
        }

        private load() {
            this.rulesService
                .get()
                .then(data => this.rules = data);
        }

        public add() {
            this.rulesService
                .new(this.move, this.kills)
                .then(() => {
                    this.move = '';
                    this.kills = '';

                    this.load()
                });
        }

        public remove(rule: Models.Rule) {
            this.rulesService
                .remove(rule)
                .then(() => this.load());
        }
    }
    angular.module(Constants.ApplicationName).controller(Constants.Controllers.RulesController, RulesController);
}
